apps/web/src/modules/create-proposal/components/TransactionForm/Migration/prepareMigrationDeploy.ts

```diff
@@ -0,0 +1,190 @@
+import { DaoStoreProps } from 'src/modules/dao'
+import { ChainStoreProps } from 'src/stores/useChainStore'
+import { CHAIN_ID } from 'src/typings'
+import { unpackOptionalArray } from 'src/utils/helpers'
+
+export async function prepareMigrationDeploy(
+  targetChainId: CHAIN_ID,
+  currentChain: ChainStoreProps,
+  currentDao: DaoStoreProps,
+  zeroFounder: {
+    wallet: `0x${string}`
+    ownershipPct: bigint
+    vestExpiry: bigint
+  },
+  merkleRoot: `0x${string}`
+) {
+  const { treasury, auction, token, metadata, governor } = currentDao.addresses
+  const chain = currentChain.chain
+  const tokenContractParams = {
+    abi: tokenAbi,
+    address: token as Address,
+    chainId: chain.id,
+  }
+
+  const metadataContractParams = {
+    abi: metadataAbi,
+    address: metadata as Address,
+    chainId: chain.id,
+  }
+
+  const auctionContractParams = {
+    abi: auctionAbi,
+    address: auction as Address,
+    chainId: chain.id,
+  }
+
+  const governorContractParams = {
+    abi: governorAbi,
+    address: governor as Address,
+    chainId: chain.id,
+  }
+
+  const treasuryContractParams = {
+    abi: treasuryAbi,
+    address: treasury as Address,
+    chainId: chain.id,
+  }
+
+  const contractData = await readContracts({
+    allowFailure: false,
+    contracts: [
+      { ...tokenContractParams, functionName: 'name' },
+      { ...tokenContractParams, functionName: 'symbol' },
+      { ...tokenContractParams, functionName: 'getFounders' },
+      { ...tokenContractParams, functionName: 'totalSupply' },
+      { ...metadataContractParams, functionName: 'contractImage' },
+      { ...metadataContractParams, functionName: 'description' },
+      { ...metadataContractParams, functionName: 'contractURI' },
+      { ...auctionContractParams, functionName: 'duration' },
+      { ...auctionContractParams, functionName: 'reservePrice' },
+      { ...governorContractParams, functionName: 'votingDelay' },
+      { ...governorContractParams, functionName: 'votingPeriod' },
+      { ...governorContractParams, functionName: 'proposalThresholdBps' },
+      { ...governorContractParams, functionName: 'quorumThresholdBps' },
+      { ...governorContractParams, functionName: 'vetoer' },
+      { ...treasuryContractParams, functionName: 'delay' },
+    ] as const,
+  })
+
+  const [
+    name,
+    symbol,
+    existingFounders,
+    totalSupply,
+    daoImage,
+    description,
+    contractURI,
+    duration,
+    reservePrice,
+    votingDelay,
+    votingPeriod,
+    proposalThresholdBps,
+    quorumThresholdBps,
+    vetoer,
+    timelockDelay,
+  ] = unpackOptionalArray(contractData, 13)
+
+  const founderParams = existingFounders
+    ? [
+        zeroFounder,
+        ...existingFounders.map((x) => ({
+          wallet: x.wallet,
+          ownershipPct: BigInt(x.ownershipPct),
+          vestExpiry: BigInt(x.vestExpiry),
+        })),
+      ]
+    : [zeroFounder]
+
+  const abiCoder = new ethers.utils.AbiCoder()
```

Comment:
mentioned this to you already but remove any instances of ethers. commenting here so we dont forget